content-services/content-sources-backend
# Fixes 1868: Admin Task API

## Summary

Adds backend API for admin task interface that allows a user to view all tasks across all organizations/accounts. It is implemented as a feature that can be enabled/disabled and restricted to specific accounts in the config.

The API allows filtering by org_id, account_id, and statuses. 

The api struct has a copy of a zest struct in order to properly convert the pulp data to JSON.

The endpoint is not included in the OpenAPI docs.

## Testing steps

Add 
```
features:
  admin_tasks:
    enabled: true
    users: ["adminUser"]
```
to the config.yaml

To see whether the feature is enabled/accessible, you can make a request to the features API:

```
curl localhost:8000/api/content-sources/v1.0/features/  -H "`./scripts/header.sh 1 adminUser`"
```

To list tasks:
```
curl "http://localhost:8000/api/content-sources/v1/admin/tasks/?offset=0&limit=20"  -H "`./scripts/header.sh 1 adminUser`" 
```

To fetch a task:
```
curl "http://localhost:8000/api/content-sources/v1/admin/tasks/{uuid}"  -H "`./scripts/header.sh 1 adminUser`" 
```
